### PR TITLE
[react-s-alert]Add new package

### PR DESCRIPTION
**Pull request by mitsuruog**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/react-s-alert/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    offset?: number;
+    stack?: boolean | SAlertStackProps;
+    effect?: string;
+    beep?: string | boolean | SAlertBeepProps;
+    timeout?: string | number;
+    html?: boolean;
+    onClose?: () => void;
+    onShow?: () => void;
+    customFields?: object;
+    contentTemplate?: Function;
```
Comment (DanielRosenwasser) [inline types/react-s-alert/index.d.ts]: You should change this to `(...args: any[]) => any` and remove `ban-types` as a rule exception.

Alternatively, I think that you might want to use `React.ComponentType` judging from the source.

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

**Comment by RyanCavanaugh** [comment]
@mitsuruog Please address comments from the code reviewers.

File: types/react-s-alert/index.d.ts (hunk 1 above)
Comment (mitsuruog) [reply types/react-s-alert/index.d.ts]: gotcha. thanks!

**Comment by mitsuruog** [comment]
@DanielRosenwasser Thanks for reviewing. Please review again.

**Comment by RyanCavanaugh** [comment]
@DanielRosenwasser - Thanks for your review of this PR! Can you please look at the new code and update your review status if appropriate?

**Review by DanielRosenwasser — APPROVED** [review]

**Comment by DanielRosenwasser** [comment]
Thanks!

**Pull request merged**